File: PixelGymBack/src/main/java/pixelgym/model/Reserva.java
package pixelgym.model;

import com.google.cloud.Timestamp;
import com.google.cloud.firestore.annotation.PropertyName;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Reserva {

    // Si el campo se llama igual en Java que en Firebase (como uid o coste),
    // no hace falta @PropertyName.
    private String uid;
    private long coste;

    @PropertyName("nombre_actividad")
    private String nombre_actividad;

    @PropertyName("nombre_profesor")
    private String nombre_profesor;

    @PropertyName("fecha_sesion")
    private String fecha_sesion;

    @PropertyName("hora_inicio")
    private String hora_inicio;

    @PropertyName("estado_reserva")
    private String estado_reserva;

    @PropertyName("id_reserva")
    private String id_reserva;

    @PropertyName("id_sesion_reservada")
    private String id_sesion_reservada;

    @PropertyName("mes_anio")
    private String mes_anio;

    @PropertyName("imagen_url")
    private String imagen_url;

    @PropertyName("fecha_creacion_reserva")
    private Timestamp fecha_creacion_reserva;
}

File: PixelGym2/app/src/main/java/com/example/aplicacion/model/Reserva.kt
package com.example.aplicacion.model

import java.security.Timestamp

data class Reserva(
    val id_reserva: String = "",
    val id_sesion_reservada: String = "",
    val uid: String = "",
    val nombre_actividad: String = "",
    val nombre_profesor: String = "", // <--- AÑADIDO
    val fecha_sesion: String = "",
    val hora_inicio: String = "",
    val mes_anio: String = "",
    val estado_reserva: String = "ACTIVA",
    val fecha_creacion_reserva: Timestamp? = null
)